sdcm/nemesis.py

```diff
@@ -2800,43 +2800,57 @@ def execute_data_validation_thread(command_template, keyspace_name, number_of_ro
-        restore_task.wait_and_get_final_status(step=30, timeout=chosen_snapshot_info["expected_timeout"])
-        assert restore_task.status == TaskStatus.DONE, f'Data restoration of {chosen_snapshot_tag} has failed!'
-
-        manager_version = mgr_cluster.sctool.parsed_client_version
-        if manager_version < LooseVersion("3.2"):
-            mgr_task = mgr_cluster.create_repair_task()
-            task_final_status = mgr_task.wait_and_get_final_status(timeout=chosen_snapshot_info["expected_timeout"])
-            assert task_final_status == TaskStatus.DONE, 'Task: {} final status is: {}.'.format(
-                mgr_task.id, str(mgr_task.status))
-
-        confirmation_stress_template = persistent_manager_snapshots_dict[cluster_backend]["confirmation_stress_template"]
-        stress_queue = execute_data_validation_thread(command_template=confirmation_stress_template,
-                                                      keyspace_name=chosen_snapshot_info["keyspace_name"],
-                                                      number_of_rows=chosen_snapshot_info["number_of_rows"])
-        for stress in stress_queue:
-            self.tester.verify_stress_thread(cs_thread_pool=stress)
+            restore_task.wait_and_get_final_status(step=30, timeout=chosen_snapshot_info["expected_timeout"])
+            assert restore_task.status == TaskStatus.DONE, f'Data restoration of {chosen_snapshot_tag} has failed!'
+
+            manager_version = mgr_cluster.sctool.parsed_client_version
+            if manager_version < LooseVersion("3.2"):
+                mgr_task = mgr_cluster.create_repair_task()
+                task_final_status = mgr_task.wait_and_get_final_status(timeout=chosen_snapshot_info["expected_timeout"])
+                assert task_final_status == TaskStatus.DONE, 'Task: {} final status is: {}.'.format(
+                    mgr_task.id, str(mgr_task.status))
+
+            confirmation_stress_template = (
+                persistent_manager_snapshots_dict)[cluster_backend]["confirmation_stress_template"]
+            stress_queue = execute_data_validation_thread(command_template=confirmation_stress_template,
+                                                          keyspace_name=chosen_snapshot_info["keyspace_name"],
+                                                          number_of_rows=chosen_snapshot_info["number_of_rows"])
+            for stress in stress_queue:
+                self.tester.verify_stress_thread(cs_thread_pool=stress)
+        finally:
+            if chosen_snapshot_info:
+                if "keyspace_name" in chosen_snapshot_info.keys():
+                    keyspace = chosen_snapshot_info["keyspace_name"]
+                    InfoEvent(message=f'Removing test {keyspace=}', severity=Severity.WARNING)
+                    with self.cluster.cql_connection_patient(self.target_node) as session:
+                        session.execute(f'DROP KEYSPACE IF EXISTS "{keyspace}"')
```

Comment:
keyspace will be deleted from scylla, but data files will remain in snapshots dir, taking space.
Do we care about it?

Reply:
Yes, nemesis shouldn't be leaving snapshots

Reply:
Let's open an issue for that and solve it on different PR.

Reply:
https://github.com/scylladb/scylla-cluster-tests/issues/6830